src/backend/partaj/core/views/common.py

```diff
@@ -3,20 +3,185 @@
 """
 import codecs
 import csv
+import json
 import mimetypes
 
 from django.contrib.auth.mixins import LoginRequiredMixin
 from django.db.models import DateTimeField, Exists, ExpressionWrapper, F, OuterRef
 from django.http import FileResponse, HttpResponse
 from django.shortcuts import redirect
+from django.utils import dateformat
 from django.utils.translation import gettext as _
 from django.views import View
 from django.views.generic import TemplateView
 
+import docx
+
 from .. import models
 from ..models import ReferralAnswerAttachment, ReferralAttachment, ReferralState
 
 
+class ExportReferralView(LoginRequiredMixin, View):
+    """
+    Return one referral as a word file to authenticated users.
+    """
+
+    list_type = {"bullet_list": "List Bullet", "ordered_list": "List Number"}
+
+    def __init__(self, *args, **kwargs):
+        """
+        Override init
+        """
+        self.doc_word = docx.Document()
+        super().__init__(*args, **kwargs)
+
+    def transform_richtext(self, text):
```

Comment:
I think this should live in a separate module rather than be done by the view. Just takes a ProseMirror formatted JSON and returns docx rich text